website/source/upgrade-guides/0-7.html.markdown

```diff
@@ -0,0 +1,135 @@
+---
+layout: "downloads"
+page_title: "Upgrading to Terraform 0.7"
+sidebar_current: "upgrade-guides-0-7"
+description: |-
+  Upgrading to Terraform v0.7
+---
+
+# Upgrading to Terraform v0.7
+
+Terraform v0.7 is a major release, and thus includes some backwards incompatibilities that you'll need to consider when upgrading. This guide is meant to help with that process.
+
+The goal of this guide is to cover the most common upgrade concerns and issues that would benefit from more explanation and background. The exhaustive list of changes will always be the [Terraform Changelog](https://github.com/hashicorp/terraform/blob/master/CHANGELOG.md). After reviewing this guide, review the Changelog to check on specific notes about the resources and providers you use.
+
+## Plugin Binaries
+
+Before v0.7, Terraform's built-in plugins for providers and provisioners were each distributed as separate binaries.
+
+```
+terraform               # core binary
+terraform-provider-*    # provider plugins
+terraform-provisioner-* # provisioner plugins
+```
+
+These binaries needed to all be extracted to somewhere on your `$PATH` for Terraform to work.
+
+As of v0.7, these plugins all ship embedded in a single binary. This means that if you just extract the v0.7 archive into a path, you may still have the old separate binaries in your `$PATH`. You'll need to remove them manually.
+
+For example, if you keep Terraform binaries in `/usr/local/bin` you can clear out the old external binaries like this:
+
+```
+rm /usr/local/bin/terraform-*
+```
+
+External plugin binaries continue to work using the same pattern, but due to updates to the RPC protocol, they will need to be recompiled to be compatible with Terraform v0.7.
+
+## Maps in Displayed Plans
+
+When displaying a plan, Terraform now distinguishes attributes of type map by using a `%` character for the "length field".
+
+Here is an example showing a diff that includes both a list and a map:
+
+```
+somelist.#:  "0" => "1"
+somelist.0:  "" => "someitem"
+somemap.%:   "0" => "1"
+somemap.foo: "" => "bar"
+```
+
+## Interpolation Changes
+
+There are a few changes to Terraform's interpolation language that may require updates to your configs.
+
+### String Concatenation
+
+The `concat()` interpolation function used to work for both lists and strings. It now only works for lists.
+
+```
+"${concat(var.foo, "-suffix")}"     # => Error! No longer supported.
+```
+
+Instead, you can use nested interpolation braces for string concatenation.
+
+```
+"${"${var.foo}-suffix"}"
```

Comment:
This is kind of ugly - `attribute_name = "${var.foo}-suffix"` should be sufficient without need for nesting braces